tests/draft2019-09/pattern.json

```diff
@@ -55,5 +55,62 @@
                 "valid": true
             }
         ]
+    },
+    {
+        "description": "Meta-schema validation tests: strings",
+        "schema": {
+            "$ref": "https://json-schema.org/draft/2019-09/schema"
+        },
+        "tests": [
+            {
+                "description": "Non-string pattern: number",
+                "data": {
+                    "pattern": 1
+                },
+                "valid": false
+            },
+            {
+                "description": "Non-string pattern: Boolean",
+                "data": {
+                    "pattern": true
+                },
+                "valid": false
+            },
+            {
+                "description": "Non-string pattern: array",
+                "data": {
+                    "pattern": []
+                },
+                "valid": false
+            },
+            {
+                "description": "Non-string pattern: object",
+                "data": {
+                    "pattern": {}
+                },
+                "valid": false
+            },
+            {
+                "description": "Non-string pattern: null",
+                "data": {
+                    "pattern": null
+                },
+                "valid": false
+            },
+            {
+                "description": "Invalid pattern",
+                "data": {
+                    "pattern": "("
```

Comment:
This only fails to validate if formats are enabled as validators, not just annotations, which is not the default configuration -- consequently this should be moved into the optional/formats directory.

Reply:
I'm not sure what you want here. "pattern" isn't the same thing as "format".

Reply:
It appears, from your comments, that you're using your own validator to run the tests. That's fine, but if you're not already, try the others, because they may treat "pattern" differently, i.e. not as optional, which is what your suggestion to move to optional/formats seems to suggest.

Reply:
You are validating against this in the metaschema:

```
    "properties": {
        ...
        "pattern": {
            "type": "string",
            "format": "regex"
        },
        ...
}
```
There is nothing to cause a failure here unless `"format":"regex"` is treated as a validation assertion.

Reply:
Some references: https://json-schema.org/draft/2019-09/json-schema-validation.html#rfc.section.3 says that [Section 7 (format)](https://json-schema.org/draft/2019-09/json-schema-validation.html#format) and [Section 8 (content)](https://json-schema.org/draft/2019-09/json-schema-validation.html#content) can be treated as only annotations and it doesn't mention anything about ["pattern"](https://json-schema.org/draft/2019-09/json-schema-validation.html#rfc.section.6.3.3), which is in Section 6.

Reply:
The issue is not "pattern", but "format".  "pattern" is the property being validated. The keyword it is being validated against is "format".

[section 7.2.2](https://json-schema.org/draft/2019-09/json-schema-validation.html#rfc.section.7.2.2) says "When the vocabulary is declared with a value of false, an implementation... MUST NOT evaluate "format" as an assertion unless it is explicitly configured to do so", which is exactly why some tests are split off into the optional/format directory, and no format-as-assertion tests exist in the main test files.

Reply:
You're right about this. I must have, in one of my rebases, included this here. I'll move them to a different PR.

Reply:
Ok, I was confused. I actually want this to be here, but I'll exclude from this set of tests until we resolve how to test un-testable schema words because of things like "must be normalized" and "format is optional".

Reply:
Leaving it out would be fine as well, although it would be fair to be included in optional/format/regex.json -- it should be as possible (or not) to validate as the other invalid case there.

Reply:
That's not a bad suggestion, but then it's not testing the "pattern" keyword specifically, which is the explicit intent of those tests, even though it may use "format=regex" under the covers; that's why I'm hesitant to put it in the "format"-specific tests. It would be my preference to just leave this for a future PR pending some more discussion.